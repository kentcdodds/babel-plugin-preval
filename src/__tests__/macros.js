import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-macros'

const projectRoot = path.join(__dirname, '../../')

expect.addSnapshotSerializer({
  print(val) {
    return val.split(projectRoot).join('<PROJECT_ROOT>/')
  },
  test(val) {
    return typeof val === 'string'
  },
})

pluginTester({
  plugin,
  snapshot: true,
  tests: withFilename([
    {
      title: 'as tag',
      code: `
        import preval from '../macro'

        const x = preval\`module.exports = require('./fixtures/compute-one')\`
      `,
    },
    {
      title: 'as function',
      code: `
        const myPreval = require('../macro')

        const x = myPreval(\`
          module.exports = require('./fixtures/identity')({sayHi: () => 'hi'})
        \`)
      `,
    },
    {
      title: 'as jsx',
      code: `
        const Preval = require('../macro')

        const ui = (
          <Preval>
            const fs = require('fs')
            module.exports = fs.readFileSync(require.resolve('./fixtures/fixture1.md'), 'utf8')
          </Preval>
        )
      `,
    },
  ]),
})

/*
 * This adds the filename to each test so you can do require/import relative
 * to this test file.
 */
function withFilename(tests) {
  return tests.map(t => {
    const test = {babelOptions: {filename: __filename}}
    if (typeof t === 'string') {
      test.code = t
    } else {
      Object.assign(test, t)
      test.babelOptions.parserOpts = test.babelOptions.parserOpts || {}
    }
    Object.assign(test.babelOptions.parserOpts, {
      // add the jsx plugin to all tests because why not?
      plugins: ['jsx'],
    })
    return test
  })
}
