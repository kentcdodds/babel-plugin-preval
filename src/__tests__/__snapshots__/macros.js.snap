// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. as tag 1`] = `

import preval from '../macro'

const x = preval\`module.exports = require('./fixtures/compute-one')\`

      ↓ ↓ ↓ ↓ ↓ ↓

const x = 1;

`;

exports[`2. as function 1`] = `

const myPreval = require('../macro')

const x = myPreval(\`
  module.exports = require('./fixtures/identity')({sayHi: () => 'hi'})
\`)

      ↓ ↓ ↓ ↓ ↓ ↓

const x = { "sayHi": function sayHi() {
    return 'hi';
  } };

`;

exports[`3. as jsx 1`] = `

const Preval = require('../macro')

const ui = (
  <Preval>
    const fs = require('fs')
    module.exports = fs.readFileSync(require.resolve('./fixtures/fixture1.md'), 'utf8')
  </Preval>
)

      ↓ ↓ ↓ ↓ ↓ ↓

const ui = <div>"# fixture\\n\\nThis is some file thing...\\n"</div>;

`;
